src/app/wallet/proposals/proposal-details/proposal-details.component.ts

```diff
@@ -0,0 +1,151 @@
+import { Component, Input, OnInit, OnDestroy } from '@angular/core';
+import { MatDialog } from '@angular/material';
+import * as d3 from 'd3';
+import { Log } from 'ng2-logger';
+
+import {
+  ProposalVoteConfirmationComponent
+} from 'app/modals/proposal-vote-confirmation/proposal-vote-confirmation.component';
+import { ProposalsService } from 'app/wallet/proposals/proposals.service';
+import { SnackbarService } from 'app/core/snackbar/snackbar.service';
+import { VoteOption } from 'app/wallet/proposals/models/vote-option.model';
+import { ModalsHelperService } from 'app/modals/modals-helper.service';
+import { Proposal } from 'app/wallet/proposals/models/proposal.model';
+import { ProposalResult } from 'app/wallet/proposals/models/proposal-result.model';
+import { GraphOption } from 'app/wallet/proposals/models/proposal-result-graph-option.model';
+import { VoteDetails } from 'app/wallet/proposals/models/vote-details.model';
+
+@Component({
+  selector: 'app-proposal-details',
+  templateUrl: './proposal-details.component.html',
+  styleUrls: [
+    './proposal-details.component.scss',
+    './../../../../assets/css/nvd3/nv.d3.css'
```

Comment:
I guess this works, but still does not seem to be a very clean way of doing this. 
Angular css imports can be done by using `@import` as well: see https://angular.io/guide/component-styles#css-imports
Using this method, along with the `~` character, would an import statement such as `@import '~/nvd3/build/nv.d3.css'` above the component definition not work?

Reply:
Yeah, we can move that CSS file `path` in the component.scss file as well.
`
@import "assets/css/nvd3/nv.d3.css";
`
OR
`
@import "./../../../../assets/css/nvd3/nv.d3.css";
`

what you refer to?

Reply:
@zaSmilingIdiot 
`@import '~/nvd3/build/nv.d3.css' above the component definition not work?`

working but it refer to the `node_module/nvd3/build/nv.d3.css` file as that is not a ideal case to use the node_module file directly?

Reply:
It depends... If the styles are copied out of the node_modules package, then you lose any future updates to the styles when updating the packages. If that is the intention (ie: fix the styles used until a manual update is done int he future), then copying the css out to a local dir is better.

My original comment on this was with the context that specifying imports, particularly for node_modules packages should preferably not make use of relative folder locations.

Specifying '../../../node_modules/path/to/package'` as an import path for a node_modules package is not great, as it means that if the component is ever moved, then the relative path for that file needs to be updated. Using `@import '~/nvd3/build/nv.d3.css'` would be preferable since moving the component around in your project hierarchy (eg: during a refactor), then no updates to import paths are necessary. 
For the 'stylesUrl' meta in `@Component`, those should remain as being relative, since those are typically located in the same folder as the component definition, and thus when moving a component (the containing folder would be moved), and so the relative styles would be moved as well.

Reply:
After chatting with @cis-shubham-t it looks like webpack is having a problem with this type of import. Will look into this, but in the meanwhile, I guess this can be left as is for now.